localstack/services/sqs/provider.py

```diff
@@ -1177,10 +1193,20 @@ def send_message_batch(
 
         successful = []
         failed = []
+        # added to address maximum batch message size (issue #6740)
+        batch_message_size = 0
+        max_message_size = queue.default_attributes()[QueueAttributeName.MaximumMessageSize]
+        max_message_size = int(max_message_size)
 
         with queue.mutex:
             for entry in entries:
                 try:
+                    # must convert to unicode to get actual bytes
+                    batch_message_size += len(entry.get("MessageBody").encode("utf8"))
+                    if batch_message_size > max_message_size:
+                        error = "Invalid batch message size found. Valid message size 262,144 bytes = 256KiB"
+                        raise InvalidMessageContents(error)
```

Comment:
My thought on that is with the loop, it would have to concatenate the message bodies of all batch messages. I chose to increment batch_message_size instead to avoid the overhead of string concatenation. String concatenation would definitely make for cleaner code.

Reply:
that's a fair point :-) feel free to leave it.